hypyp/prep.py

```diff
@@ -166,29 +179,30 @@ def ICA_fit(epochs: list, n_components: int, method: str, fit_params: dict, rand
         icas: list of Independant Components for each participant (IC are MNE
             objects, see MNE documentation for more details).
     """
-    icas = []
-    for epoch in epochs:
+    icas: List[ICA] = []
+    for epochs_subj in epochs:
         # per subj
         # applying AR to find global rejection threshold
-        reject = get_rejection_threshold(epoch, ch_types='eeg')
+        reject = get_rejection_threshold(epochs_subj, ch_types='eeg')
         # if very long, can change decim value
-        print('The rejection dictionary is %s' % reject)
+        print(f"The rejection dictionary is {reject}")
 
         # fitting ICA on filt_raw after AR
         ica = ICA(n_components=n_components,
                   method=method,
                   fit_params= fit_params,
-                  random_state=random_state).fit(epoch)
```

Comment:
I might be wrong, but I think calling `.fit(epoch)` here is useless, since we call `.fit()` below again.